robot_radius: 0.25  # distance a circular robot should be clear of the obstacle
# footprint: [[x0, y0], [x1, y1], ... [xn, yn]]  # if the robot is not circular

map_type: costmap

static_layer:
  enabled: true
  unknown_cost_value: -1
  lethal_cost_threshold: 100
  map_topic: /map
  first_map_only: false #when amcl, it can set "true"
  track_unknown_space: true
  use_maximum: false
  trinary_costmap: true
  
obstacle_layer:
  enabled: true 
  #Sensor management parameters
  observation_sources: laser_scan_sensor
  laser_scan_sensor:
    topic: /scan
    sensor_frame: laser_link
    #observation_persistence: 0.0
    #expected_update_rate: 0.0
    data_type: LaserScan
    clearing: true #false
    marking: true
    #max_obstacle_height: 0.3
    #min_obstacle_height: 0.1
    #obstacle_range: 2.5
    #raytrace_range: 3.0
    #inf_is_valid: false
  
  #Global Filtering Parameters
  #max_obstacle_height: 0.6
  obstacle_range: 2.5
  raytrace_range: 3.0
  
  #ObstacleCostmapPlugin
  track_unknown_space: true #false
  footprint_clearing_enabled: true
  
  #VoxelCostmapPlugin
  #origin_z: 0.0
  #z_resolution: 0.2
  #z_voxels: 10
  #unknown_threshold: 15
  #mark_threshold: 0
  #publish_voxel_map: true
  #footprint_clearing_enabled: true
  
global_inflation_layer:
  enabled: true
  inflation_radius: 0.5 #0.55
  cost_scaling_factor: 2.0 #10.0
  
local_inflation_layer:
  enabled: true
  inflation_radius: 0.2 #0.55
  cost_scaling_factor: 5.0 #10.0
  
#sonar_layer:
  #topic: /sonar
  #no_readings_timeout: 0.0
  #clear_threshold: 2
  #mark_threshold: 8
  #clear_on_max_reading: false
   
polygon_layer:
  resize_to_polygon: true
  

